### DISCUSSION TITLE
Generating values on multiple dependent fields

### DISCUSSION THREAD

**Discussion by MathieuLegault1** [instancio/instancio#1053] [Q&A] [answered]
Hi,

I'm currently trying to create a new object where one field needs to be generated using two (or more) dependent fields. I know I can use assign when there is only one dependent field, but it doesn't seem to work when there are multiple dependencies.

For example:
```java
class Person {
  String firstName;
  String lastName;
  String fullName;
}
```
What I want is to compute the fullName after firstName and lastName are generated. I'd prefer not to use onComplete if possible, as my object is immutable.

Perhaps there is another way apart from assign, but I haven't been able to think of one. Any alternatives I can come up with lose the randomness of the data. Like generating my fields outside of the generate/supplier.

Thanks!

**Comment by armandino** [comment, marked as answer]
@MathieuLegault1 it's true that `assign()` does not support generating values that depend on multiple fields. In this case though, it should be possible to specify the entire `Person` object and use the `as(Function)` to calculate the full name:

```java
Assignment fullName = Assign
    .valueOf(Person.class)
    .to(Person::getFullName)
    .as(((Person p) -> p.getFirstName() + " " + p.getLastName()));

Person person = Instancio.of(Person.class)
    .assign(fullName)
    .create();
```

Another option is to generate each name field separately:

```java
String firstName = Instancio.create(String.class);
String lastName = Instancio.create(String.class);

Person person = Instancio.of(Person.class)
    .set(field(Person::getFirstName), firstName)
    .set(field(Person::getLastName), lastName)
    .set(field(Person::getFullName), firstName + " " + lastName)
    .create();
```

**Reply by MathieuLegault1** [reply]
That is exactly what I want !

Thanks !
